### PR TITLE
chore: Add #[recursive] Explainer to Documentation

### PR DESCRIPTION
**Pull request by paulallensuxs**
# Description

## Problem

Step towards [#4392](https://github.com/noir-lang/noir/issues/4392)

The documentation for Recursive Proofs was lacking information on the newly introduced `#[recursive]` attribute, which is crucial for understanding how to mark circuits for recursive proof generation. 

## Summary

This pull request updates the Recursive Proofs documentation page to include a comprehensive section on the `#[recursive]` attribute. It explains the attribute's purpose, how it should be used, and provides an example demonstrating its application within a circuit definition. 

## Additional Context

The introduction of the `#[recursive]` attribute simplifies the process of designating circuits for recursive proofs, eliminating the need for manual flagging in the tooling infrastructure. 

## Documentation

- [ ] No documentation needed.
- [x] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by signorecello — CHANGES_REQUESTED** [review]
Thanks @paulallensuxs I'd just add this change to the `versioned_docs`, namely under the `v0.24.0` version as specified in the issue.

This will make the doc change available for 0.24.0 which is when the change was introduced.

**Review by signorecello — APPROVED** [review]
Thanks!

File: docs/docs/noir/standard_library/recursion.md [hunk 1]
```diff
@@ -8,6 +8,26 @@ Noir supports recursively verifying proofs, meaning you verify the proof of a No
...
+#[recursive]
+fn main(x: Field, y: pub Field) {
+    assert(x == y, "x and y are not equal");
+}
+
+// This marks the circuit as recursion-friendly and indicates that proofs generated from this circuit
+// are intended for recursive verification.
+```
+
+By incorporating this attribute directly in the circuit's definition, tooling like Nargo and NoirJS can automatically handle the proving of recursive Noir programs without requiring additional flags or configurations.
```
Comment (Savio-Sou) [inline docs/docs/noir/standard_library/recursion.md:27]:
```suggestion
By incorporating this attribute directly in the circuit's definition, tooling like Nargo and NoirJS can automatically execute recursive-specific duties for Noir programs (e.g. recursive-friendly proof artifact generation) without additional flags or configurations.
```

We are currently aiming for automation in artifacts generation, but not necessarily proving + recursive verification yet as we would need a generalized workflow that could support different forms of recursive proving.

Additional context: https://github.com/noir-lang/noir/issues/4336#issuecomment-1938767144

File: docs/versioned_docs/version-v0.24.0/noir/standard_library/recursion.md [hunk 2]
```diff
@@ -8,6 +8,26 @@ Noir supports recursively verifying proofs, meaning you verify the proof of a No
...
+#[recursive]
+fn main(x: Field, y: pub Field) {
+    assert(x == y, "x and y are not equal");
+}
+
+// This marks the circuit as recursion-friendly and indicates that proofs generated from this circuit
+// are intended for recursive verification.
+```
+
+By incorporating this attribute directly in the circuit's definition, tooling like Nargo and NoirJS can automatically handle the proving of recursive Noir programs without requiring additional flags or configurations.
```
Comment (Savio-Sou) [inline docs/versioned_docs/version-v0.24.0/noir/standard_library/recursion.md:27]:
```suggestion
By incorporating this attribute directly in the circuit's definition, tooling like Nargo and NoirJS can automatically execute recursive-specific duties for Noir programs (e.g. recursive-friendly proof artifact generation) without additional flags or configurations.
```

**Review by Savio-Sou — COMMENTED** [review]
Thank you for submitting the PR!

Minor nits:

**Pull request merged**